src/Evolution/DiscontinuousGalerkin/InitializeElement.hpp

```diff
@@ -85,151 +84,238 @@ namespace Actions {
+
+    using compute_tags = db::AddComputeTags<
+        Tags::Time,
+        Tags::deriv<typename variables_tag::tags_list,
+                    typename System::gradients_tags,
+                    Tags::InverseJacobian<Tags::ElementMap<Dim>,
+                                          Tags::LogicalCoordinates<Dim>>>>;
+
+    template <typename TagsList, typename Metavariables>
+    static auto initialize(
+        db::DataBox<TagsList>&& box,
+        const Parallel::ConstGlobalCache<Metavariables>& cache,
+        const Time& initial_time, const TimeDelta& initial_dt) noexcept {
+      using Vars = typename variables_tag::type;
+      using DtVars = typename dt_variables_tag::type;
+
+      const auto& time_stepper = Parallel::get<CacheTags::TimeStepper>(cache);
 
-    // Set initial data from analytic solution
-    Variables<variables_tags> vars{num_grid_points};
-    vars.assign_subset(Parallel::get<solution_tag>(cache).variables(
-        inertial_coords, initial_time.value(), variables_tags{}));
-
-    typename Tags::HistoryEvolvedVariables<
-        Tags::Variables<variables_tags>,
-        Tags::dt<Tags::Variables<db::wrap_tags_in<Tags::dt, variables_tags>>>>::
-        type history_dt_vars;
-    if (not time_stepper.is_self_starting()) {
-      // We currently just put initial points at past slab boundaries.
-      Time past_t = initial_time;
-      TimeDelta past_dt = initial_dt;
-      for (size_t i = time_stepper.number_of_past_steps(); i > 0; --i) {
-        past_dt = past_dt.with_slab(past_dt.slab().advance_towards(-past_dt));
-        past_t -= past_dt;
-        Variables<variables_tags> hist_vars{num_grid_points};
-        Variables<db::wrap_tags_in<Tags::dt, variables_tags>> dt_hist_vars{
-            num_grid_points};
-        hist_vars.assign_subset(Parallel::get<solution_tag>(cache).variables(
-            inertial_coords, past_t.value(), variables_tags{}));
-        dt_hist_vars.assign_subset(Parallel::get<solution_tag>(cache).variables(
-            inertial_coords, past_t.value(),
-            db::wrap_tags_in<Tags::dt, variables_tags>{}));
-        history_dt_vars.insert_initial(past_t, std::move(hist_vars),
-                                       std::move(dt_hist_vars));
+      const TimeId time_id(initial_dt.is_positive(), 0, initial_time);
+      TimeId next_time_id = time_stepper.next_time_id(time_id, initial_dt);
+      if (next_time_id.is_at_slab_boundary()) {
+        const auto next_time = next_time_id.step_time();
+        next_time_id = TimeId(
+            initial_dt.is_positive(), 1,
+            next_time.with_slab(next_time.slab().advance_towards(initial_dt)));
       }
+
+      const size_t num_grid_points = db::get<Tags::Extents<Dim>>(box).product();
+      const auto& inertial_coords =
+          db::get<Tags::Coordinates<Tags::ElementMap<Dim>,
+                                    Tags::LogicalCoordinates<Dim>>>(box);
+
+      // Will be overwritten before use
+      DtVars dt_vars{num_grid_points, 0.0};
```

Comment:
It the 0's will be overwritten, should it be initialised with signalling nans? Is this the default behavious for variables?

Reply:
yes, without the `0.0`  it would be filled with signalling_nans